dspace-api/src/main/java/org/dspace/app/profile/ResearcherProfile.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.profile;
+
+import static org.dspace.core.Constants.READ;
+import static org.dspace.eperson.Group.ANONYMOUS;
+
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import org.dspace.content.Item;
+import org.dspace.content.MetadataValue;
+import org.dspace.util.UUIDUtils;
+import org.springframework.util.Assert;
+
+/**
+ * Object representing a Researcher Profile.
+ *
+ * @author Luca Giamminonni (luca.giamminonni at 4science.it)
+ *
+ */
+public class ResearcherProfile {
+
+    private final Item item;
+
+    private final MetadataValue dspaceObjectOwner;
+
+    /**
+     * Create a new ResearcherProfile object from the given item.
+     *
+     * @param  item                     the profile item
+     * @throws IllegalArgumentException if the given item has not a dspace.object.owner
+     *                                  metadata with a valid authority
+     */
+    public ResearcherProfile(Item item) {
+        Assert.notNull(item, "A researcher profile requires an item");
+        this.item = item;
+        this.dspaceObjectOwner = getDspaceObjectOwnerMetadata(item);
+    }
+
+    public UUID getId() {
+        return UUIDUtils.fromString(dspaceObjectOwner.getAuthority());
+    }
+
+    public String getFullName() {
```

Comment:
Copying feedback from https://github.com/DSpace/DSpace/pull/8263

Best to mention in the JavaDocs that this name can be outdated, and the latest value should be retrieved from the EPerson instead. This method can best be made deprecated, or removed

Reply:
I think this change wasn't committed yet